test/integration/targets/git/tasks/missing_hostkey_acceptnew.yml

```diff
@@ -0,0 +1,77 @@
+- name: MISSING-HOSTKEY | check accept_newhostkey support
+  shell: ssh -o StrictHostKeyChecking=accept-new -V
+  register: ssh_supports_accept_newhostkey
+  ignore_errors: true
+
+- block:
+    - name: MISSING-HOSTKEY | accept_newhostkey when ssh does not support the option
+      git:
+        repo: '{{ repo_format2 }}'
+        dest: '{{ checkout_dir }}'
+        accept_newhostkey: true
+        ssh_opts: '-o UserKnownHostsFile={{ output_dir }}/known_hosts'
+      register: git_result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - git_result is failed
+          # - "'does not support' in git_result.warnings" (how to handle this?)
```

Comment:
Things like this are sometimes easier to test in unit tests because you use mocks in order to hit the desired code path.

We have CentOS 6, CentOS 7, and RHEL 7 in our test matrix, which all have an old enough version of ssh that this code path should be traversed. Add another `assert` task with a conditional based on distribution + version or the output of `command: ssh -V`. Whichever you prefer.


```yaml
when: ansible_facts.distribution + ansible_facts.distribution_major_version in ['CentOS6', 'CentOS6', 'RedHat7']`
```

```yaml
- name: Get ssh version
  command: ssh -V
  register: ssh_version_command

- assert:
     that:
       - "'does not support' in git_result.warnings" (how to handle this?)
  when: ssh_version is version('7.5', '<')
  vars:
    ssh_version: "{{ ssh_version_command.stderr.split('_')[1].split(',')[0].split('p')[0] }}"
```

Reply:
`does not support` was present in `git_result.warnings`, but it didn't work this way, I changed it in this commit: https://github.com/ansible/ansible/pull/70499/commits/d79ab1beac1d3f01e6ee22e04f9b059354cef6b8 and now it works.